src/libosd/include/osd/cl_dem_uart.h

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OSD_CL_DEM_UART_H
+#define OSD_CL_DEM_UART_H
+
+#include <stdlib.h>
+
+#include <osd/osd.h>
+#include <osd/packet.h>
+#include <osd/hostmod.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+
+/**
+ * A osd_dem_uart_event represents one or more characters.
+ */
+struct osd_dem_uart_event {
+	uint8_t num_chars;
+	uint8_t chars[0];
```

Comment:
Please adjust the datatypes:
- Use `char` for the character stream.
- Use `size_t` for the length.

You also might want to change the naming a bit: `chars` to `str`, and `num_chars` to `len`. 
Reasoning: UART DEM does not care about the encoding of the data, it only forwards a byte stream. In ASCII/Latin-1, one char == one byte. With other encodings this equation is not necessarily true. For example, a console set to UTF-8 might send two or more bytes to display a single "character."